python_modules/dagster-graphql/dagster_graphql/schema/auto_materialize_asset_evaluations.py

```diff
@@ -0,0 +1,104 @@
+
+
+class GrapheneAutoMaterializeConditionWithDecisionType(graphene.Interface):
+    decisionType = graphene.NonNull(GrapheneAutoMaterializeDecisionType)
+
+
+class GrapheneFreshnessAutoMaterializeCondition(graphene.ObjectType):
+    class Meta:
+        name = "FreshnessAutoMaterializeCondition"
+        interfaces = (GrapheneAutoMaterializeConditionWithDecisionType,)
+
+
+class GrapheneDownstreamFreshnessAutoMaterializeCondition(graphene.ObjectType):
+    class Meta:
+        name = "DownstreamFreshnessAutoMaterializeCondition"
+        interfaces = (GrapheneAutoMaterializeConditionWithDecisionType,)
+
+
+class GrapheneParentMaterializedAutoMaterializeCondition(graphene.ObjectType):
+    class Meta:
+        name = "ParentMaterializedAutoMaterializeCondition"
+        interfaces = (GrapheneAutoMaterializeConditionWithDecisionType,)
+
+
+class GrapheneMissingAutoMaterializeCondition(graphene.ObjectType):
+    class Meta:
+        name = "MissingAutoMaterializeCondition"
+        interfaces = (GrapheneAutoMaterializeConditionWithDecisionType,)
+
+
+class GrapheneParentOutdatedAutoMaterializeCondition(graphene.ObjectType):
+    class Meta:
+        name = "ParentOutdatedAutoMaterializeCondition"
+        interfaces = (GrapheneAutoMaterializeConditionWithDecisionType,)
+
+
+class GrapheneAutoMaterializeCondition(graphene.Union):
+    class Meta:
+        name = "AutoMaterializeCondition"
+        types = (
+            GrapheneFreshnessAutoMaterializeCondition,
+            GrapheneDownstreamFreshnessAutoMaterializeCondition,
+            GrapheneParentMaterializedAutoMaterializeCondition,
+            GrapheneMissingAutoMaterializeCondition,
+            GrapheneParentOutdatedAutoMaterializeCondition,
+        )
+
+
+def create_graphene_auto_materialize_condition(condition: AutoMaterializeCondition):
+    if isinstance(condition, FreshnessAutoMaterializeCondition):
+        return GrapheneFreshnessAutoMaterializeCondition(decisionType=condition.decision_type)
+    elif isinstance(condition, DownstreamFreshnessAutoMaterializeCondition):
+        return GrapheneDownstreamFreshnessAutoMaterializeCondition(
+            decisionType=condition.decision_type
+        )
+    elif isinstance(condition, ParentMaterializedAutoMaterializeCondition):
+        return GrapheneParentMaterializedAutoMaterializeCondition(
+            decisionType=condition.decision_type
+        )
+    elif isinstance(condition, MissingAutoMaterializeCondition):
+        return GrapheneMissingAutoMaterializeCondition(decisionType=condition.decision_type)
+    elif isinstance(condition, ParentOutdatedAutoMaterializeCondition):
+        return GrapheneParentOutdatedAutoMaterializeCondition(decisionType=condition.decision_type)
+    else:
+        check.failed(f"Unexpected condition type {condition}")
+
+
+class GrapheneAutoMaterializeAssetEvaluationRecord(graphene.ObjectType):
+    evaluationId = graphene.NonNull(graphene.BigInt)
```

Comment:
can we just name this `id` so that it works with apollo client's caching please 🙏🏾

Reply:
I'll expose the separate id column from the db. evaluationId is unique within a query for one asset key, but not across asset keys so internally `id` is a separate thing

Reply:
sweet thank you!